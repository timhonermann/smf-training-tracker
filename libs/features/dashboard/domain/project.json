{
  "name": "dashboard-domain",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/features/dashboard/domain/src",
  "prefix": "stt",
  "projectType": "library",
  "tags": ["domain:dashboard", "type:domain"],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint"
    }
  }
}
