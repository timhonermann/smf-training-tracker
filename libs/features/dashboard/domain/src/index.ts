export * from './lib/state/summary-metric-store';
export * from './lib/state/training-requirements-store';
