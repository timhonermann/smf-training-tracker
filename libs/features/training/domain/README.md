# training-domain

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test training-domain` to execute the unit tests.
