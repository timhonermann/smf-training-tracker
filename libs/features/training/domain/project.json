{
  "name": "training-domain",
  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/features/training/domain/src",
  "prefix": "stt",
  "projectType": "library",
  "tags": [
    "domain:training",
    "type:domain"
  ],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint"
    }
  }
}
