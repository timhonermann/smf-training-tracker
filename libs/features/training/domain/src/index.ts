export * from './lib/state/training-location-store';
export * from './lib/state/training-store';
