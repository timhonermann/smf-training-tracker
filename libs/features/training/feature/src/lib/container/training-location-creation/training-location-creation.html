<stt-back-button [link]="backLink()" />

<mat-form-field appearance="outline">
  <mat-label>Location Name</mat-label>
  <input matInput [formControl]="nameControl
" />
</mat-form-field>

<stt-button [isDisabled]="!nameControl.valid" (clicked)="create()">Create</stt-button>
