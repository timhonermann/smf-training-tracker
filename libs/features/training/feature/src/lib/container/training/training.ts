import { ChangeDetectionStrategy, Component } from '@angular/core';

@Component({
  selector: 'stt-training',
  imports: [],
  templateUrl: './training.html',
  styleUrl: './training.scss',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class Training {}
