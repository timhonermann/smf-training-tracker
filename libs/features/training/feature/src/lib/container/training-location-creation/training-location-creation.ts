import {
  ChangeDetectionStrategy,
  Component,
  inject,
  signal,
} from '@angular/core';
import { MatFormField, MatLabel } from '@angular/material/form-field';
import { FormControl, ReactiveFormsModule, Validators } from '@angular/forms';
import { MatInput } from '@angular/material/input';
import { BackButton, Button } from '@stt/shared/button/ui';
import { featureRoutes } from '@stt/shared/routing/model';
import { TrainingLocationStore } from '@stt/features/training/domain';
import { TrainingLocationCreationData } from '@stt/features/training/model';

@Component({
  selector: 'stt-training-location-creation',
  imports: [
    MatFormField,
    MatLabel,
    MatInput,
    ReactiveFormsModule,
    BackButton,
    Button,
  ],
  templateUrl: './training-location-creation.html',
  styleUrl: './training-location-creation.scss',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class TrainingLocationCreation {
  readonly #store = inject(TrainingLocationStore);

  readonly nameControl = new FormControl<string>('', [Validators.required]);

  readonly backLink = signal(
    `/${featureRoutes.TRAINING}/location`
  ).asReadonly();

  create(): void {
    const creationData: TrainingLocationCreationData = {
      name: this.nameControl.value ?? '',
    };

    this.#store.create(creationData);
  }
}
