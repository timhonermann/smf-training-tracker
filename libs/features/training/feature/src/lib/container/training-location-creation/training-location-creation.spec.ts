import { ComponentFixture, TestBed } from '@angular/core/testing';
import { TrainingLocationCreation } from './training-location-creation';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { TrainingLocationStore } from '@stt/features/training/domain';
import { describe, expect } from 'vitest';
import { MockProvider } from 'ng-mocks';
import { ReactiveFormsModule } from '@angular/forms';

describe('TrainingLocationCreation', () => {
  let component: TrainingLocationCreation;
  let fixture: ComponentFixture<TrainingLocationCreation>;
  let store: TrainingLocationStore;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [TrainingLocationCreation],
      providers: [
        MockProvider(TrainingLocationStore, {
          create: vi.fn(),
        }),
      ],
    })
      .overrideComponent(TrainingLocationCreation, {
        set: {
          imports: [ReactiveFormsModule],
          schemas: [CUSTOM_ELEMENTS_SCHEMA],
        },
      })
      .compileComponents();

    store = TestBed.inject(TrainingLocationStore);

    fixture = TestBed.createComponent(TrainingLocationCreation);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('backLink', () => {
    expect(component.backLink()).toMatchSnapshot();
  });

  describe('create', () => {
    it('should call store create with nameControl value', () => {
      // arrange
      const value = 'Pilatus';
      const createSpy = vi.spyOn(store, 'create');

      component.nameControl.setValue(value);

      // act
      component.create();

      // assert
      expect(createSpy).toHaveBeenCalledExactlyOnceWith({ name: value });
    });
  });
});
