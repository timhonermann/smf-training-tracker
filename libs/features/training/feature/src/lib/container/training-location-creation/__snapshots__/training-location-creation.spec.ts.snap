// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TrainingLocationCreation > backLink 1`] = `"/training/location"`;
